src/metrics/graphite_metric.ts

```diff
@@ -1,12 +1,43 @@
-import { AbsractMetric, Datasource, MetricId  } from "./metric";
+import { AbsractMetric, Datasource, MetricId  } from './metric';
+
+import * as moment from 'moment';
 
 export class GraphiteMetric extends AbsractMetric {
 
+    private static QUERY_TIME_REGEX = /\&from=[^\&]+\&until=[^\&]+\&/;
```

Comment:
from and until cannot be empty? Seems like it wouldn't work for `&from=&until=123&`
Also, maybe `&`s at start and end are not necessary?
Seems like you have 4 spaces instead of 2 here

Reply:
spaces fixed
regexp fixed